pkg/webhooks/controlplanemachineset/webhooks.go

```
@@ -350,11 +351,31 @@ func validateOpenShiftProviderConfig(logger logr.Logger, parentPath *field.Path,
 		return validateOpenShiftGCPProviderConfig(providerSpecPath.Child("value"), providerConfig.GCP())
 	case configv1.OpenStackPlatformType:
 		return validateOpenShiftOpenStackProviderConfig(providerSpecPath.Child("value"), providerConfig.OpenStack())
+	case configv1.NutanixPlatformType:
+		return validateOpenShiftNutanixProviderConfig(providerSpecPath.Child("value"), providerConfig.Nutanix(), template.FailureDomains)
 	}
 
 	return []error{}
 }
 
+// validateOpenShiftNutanixProviderConfig runs Nutanix specific checks on the provider config on the ControlPlaneMachineSet.
+// This ensure that the ControlPlaneMachineSet can safely replace Nutanix control plane machines.
+func validateOpenShiftNutanixProviderConfig(parentPath *field.Path, providerConfig providerconfig.NutanixProviderConfig, failureDomains *machinev1.FailureDomains) []error {
+	errs := []error{}
+
+	if failureDomains == nil {
+		return errs
+	}
+
+	for _, fdRef := range failureDomains.Nutanix {
+		if fd, _ := providerConfig.GetFailureDomainByName(fdRef.Name); fd == nil {
```

Why ignoring the error here?

---

Will fix this.

---

Fixed.